phpBB/phpbb/install/module_base.php

```diff
@@ -110,10 +110,38 @@ public function run()
 
 		if ($task_index < $iterator->count())
 		{
-			$iterator->seek($task_index);
+		    // Receiving task index
+		    $get_task_index = (int)$this->iohandler->get_input('task_index', 0);
+		    // Validation check
+		    if($get_task_index > 0 && $get_task_index < $iterator->count()){
+                $get_task_index--;
+                $iterator->seek($get_task_index);
+                // Setting the menu point as active
+                $this->iohandler->set_active_stage_menu($iterator->current()->get_navigation_stage_path());
+                $this->iohandler->send_response();
+            }
+            else{
+                // Displaying completed as checked
+                for($i = 0; $i < $task_index; $i++){
+                    $iterator->seek($i);
+                        $this->iohandler->set_finished_stage_menu($iterator->current()->get_navigation_stage_path());
+                        $this->iohandler->send_response();
+                }
+                $iterator->seek($task_index);
+                // Setting the menu point as active
+                $this->iohandler->set_active_stage_menu($iterator->current()->get_navigation_stage_path());
+                $this->iohandler->send_response();
+            }
 		}
 		else
 		{
+            // Checking all steps as completed
+            for($i = 0; $i < $iterator->count(); $i++){
```

Comment:
Yes, but why would it be set to an invalid value? The system is set by the administrator and he can change files anyways.